### ISSUE TITLE
Crash when using auto SPMD on TPU pod with more than 1 VM

### ISSUE THREAD

**Issue by dudulightricks** [pytorch/xla#7853]
## 🐛 Bug

We were not able to run auto SPMD on TPU pod with 2 VMs (v16p). We run the same program on both of the VMs in parallel.

Example code:

`import torch
import torch_xla.core.xla_model as xm
import torch_xla.distributed.parallel_loader as pl
import torch_xla.runtime as xr
from torch.distributed._tensor import DeviceMesh, distribute_module
from torch_xla.distributed.spmd import auto_policy
from torch import nn

xr.use_spmd(auto=True)

batch = {"key": torch.tensor([[1, 1], [1, 1]], device='cpu')}
dataloader = iter([batch])
dataloader_wrapper = pl.MpDeviceLoader(dataloader, device=xm.xla_device())
k = next(iter(dataloader_wrapper))["key"]

model = nn.Linear(in_features=2, out_features=2, device=xm.xla_device())
device_count = xr.global_runtime_device_count()
device_mesh = DeviceMesh("xla", list(range(device_count)))
model.to('cpu')
model = distribute_module(model, device_mesh, auto_policy)

output = model(k)
xm.mark_step()
print(output)`

1. Run the code using PJRT_DEVICE=TPU XLA_AUTO_SPMD_MESH=2,4 python code.py on each of the 2 VMs of tpu v16p.

Stacktrace: 
`*** SIGABRT received by PID 986955 (TID 987851) on cpu 39 from PID 986955; ***
E0814 18:26:54.077728  987851 coredump_hook.cc:316] RAW: Remote crash data gathering hook invoked.
E0814 18:26:54.077736  987851 coredump_hook.cc:355] RAW: Skipping coredump since rlimit was 0 at process start.
E0814 18:26:54.077742  987851 client.cc:269] RAW: Coroner client retries enabled, will retry for up to 30 sec.
E0814 18:26:54.077746  987851 coredump_hook.cc:411] RAW: Sending fingerprint to remote end.
E0814 18:26:54.077761  987851 coredump_hook.cc:420] RAW: Cannot send fingerprint to Coroner: [NOT_FOUND] stat failed on crash reporting socket /var/google/services/logmanagerd/remote_coredump.socket (Is the listener running?): No such file or directory
E0814 18:26:54.077766  987851 coredump_hook.cc:472] RAW: Dumping core locally.
F0814 18:26:54.075175  987851 auto_sharding_util.cc:1594] Check failed: tensor_shape.IsArray()
E0814 18:26:54.247832  987851 process_state.cc:805] RAW: Raising signal 6 with default behavior
Aborted (core dumped)`

Another error we got in a similar execution:
`F0813 11:57:32.451559  309636 lowering_util.cc:2882] Check failed: !data->ProducesVmreg() && !data->ProducesVreg() Unsupported conversion from vmreg/vreg to U64`

## Expected behavior

Getting the result of the execution.

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]:
 - torch_xla version: 2.4

**Comment by JackCaoG** [comment]
we never really support auto-sharding on pod because the compilation time is way too long(often time hours in larger model) and we have limited its support on single host for experimenting the shardings in small scales. Do you see the same issue when you use auto-sharding on a single host?

**Comment by dudulightricks** [comment]
No, it works on a single host. I couldn't find any documentation saying that and the error is not informative. Thanks for the reply.
